tests/math/test_matrix_manipulation.py

```diff
@@ -1068,3 +1070,189 @@ def test_expand_vector_invalid_vector(self):
+            sqrt_matrix_sparse(m)
+
+    @pytest.mark.parametrize("m, e, info", illmats_info_pairs)
+    def test_sqrt_matrix_sparse_input_ill_conditioned(self, m, e, info):
+        """Test that appropriate errors are raised for ill-conditioned matrices"""
+        m = csr_matrix(m)
+        with pytest.raises(e, match=info):
+            sqrt_matrix_sparse(m)
+
+    def test_sqrt_matrix_sparse_input_valid(self):
+        """Test that if dense input errors raised"""
+        A = np.array([[1, 0], [0, 1]])
+        with pytest.raises(TypeError, match="Only use this method for sparse matrices"):
+            sqrt_matrix_sparse(A)
+
+
+class TestDenmanBeaversIterations:
+    """Tests for the Denman-Beavers iteration method for matrix square root"""
+
+    def test_singular_matrix(self):
+        """Test that singular matrix raises appropriate error"""
+        n = 4
+        mat = csr_matrix(np.diag([0.0] + [1.0] * (n - 1)))
+
+        with pytest.raises(ValueError, match="Factor is exactly singular"):
+            _denman_beavers_iterations(mat)
+
+    def test_overflow_matrix(self):
+        """Test that matrix with very large values raises convergence warning"""
+        n = 4
+        mat = csr_matrix(np.eye(n) * 1e150)
+
+        with pytest.raises(ValueError, match="Denman Beavers not converged"):
+            _denman_beavers_iterations(mat)
+
+    def test_invalid_value_matrix(self):
+        """Test that matrix leading to invalid values raises appropriate error"""
+        n = 4
+        mat = csr_matrix(np.diag([-1e-10] + [1.0] * (n - 1)))
+
+        with pytest.raises(ValueError, match="nan or inf"):
+            _denman_beavers_iterations(mat)
+
+    def test_non_convergent_matrix(self):
+        """Test that non-convergent matrix raises appropriate error"""
+        mat = csr_matrix([[1, 1e-6], [1e-6, 1e-12 + 1e-25]])
+        with pytest.raises(ValueError, match="Denman Beavers not converged"):
+            _denman_beavers_iterations(mat)
+
+    def test_unstable_matrix(self):
+        """Test that numerically unstable matrix raises convergence warning"""
+        n = 4
+        mat = csr_matrix(np.diag([1e-200, 1e200] + [1.0] * (n - 2)))
+
+        with pytest.raises(ValueError, match="Denman Beavers not converged"):
+            _denman_beavers_iterations(mat)
+
+    @pytest.mark.parametrize("size", [2, 3, 4, 5])
+    def test_valid_positive_definite(self, size):
+        """Test that valid real, positive definite matrices work correctly"""
+        # Create a positive definite matrix
+        A = np.random.random((size, size))
+        mat = csr_matrix(np.eye(size) - 0.1 * (A @ A.T))
+
+        result = _denman_beavers_iterations(mat)
+        # Check that result is a valid square root
+        result_dense = result.toarray()
+        original_dense = mat.toarray()
+        qml.math.allclose(result_dense @ result_dense, original_dense, atol=1e-7, rtol=1e-7)
+
+    @pytest.mark.parametrize("trial", range(20))
```

Comment:
Is this something we do in the test suite? :sweat_smile:

Reply:
Yeah this is quite hacky here. I'll break them into strictly two suites later.

Reply:
Co-resolved in another suggestion